<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Strict//EN">
<html>
<head><title>C2AFAACXX_Pc2_0d_13s004125-1-1_Hejnol_lane213s004125_sequence.txt.gz FastQC Report</title>

<style type="text/css">

 @media screen {
  div.summary {
    width: 18em;
    position:fixed;
    top: 3em;
    margin:1em 0 0 1em;
  }
  
  div.main {
    display:block;
    position:absolute;
    overflow:auto;
    height:auto;
    width:auto;
    top:4.5em;
    bottom:2.3em;
    left:18em;
    right:0;
    border-left: 1px solid #CCC;
    padding:0 0 0 1em;
    background-color: white;
    z-index:1;
  }
  
  div.header {
    background-color: #EEE;
    border:0;
    margin:0;
    padding: 0.5em;
    font-size: 200%;
    font-weight: bold;
    position:fixed;
    width:100%;
    top:0;
    left:0;
    z-index:2;
  }

  div.footer {
    background-color: #EEE;
    border:0;
    margin:0;
	padding:0.5em;
    height: 1.3em;
	overflow:hidden;
    font-size: 100%;
    font-weight: bold;
    position:fixed;
    bottom:0;
    width:100%;
    z-index:2;
  }
  
  img.indented {
    margin-left: 3em;
  }
 }
 
 @media print {
	img {
		max-width:100% !important;
		page-break-inside: avoid;
	}
	h2, h3 {
		page-break-after: avoid;
	}
	div.header {
      background-color: #FFF;
    }
	
 }
 
 body {    
  font-family: sans-serif;   
  color: #000;   
  background-color: #FFF;
  border: 0;
  margin: 0;
  padding: 0;
  }
  
  div.header {
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 200%;
  font-weight: bold;
  width:100%;
  }    
  
  #header_title {
  display:inline-block;
  float:left;
  clear:left;
  }
  #header_filename {
  display:inline-block;
  float:right;
  clear:right;
  font-size: 50%;
  margin-right:2em;
  text-align: right;
  }

  div.header h3 {
  font-size: 50%;
  margin-bottom: 0;
  }
  
  div.summary ul {
  padding-left:0;
  list-style-type:none;
  }
  
  div.summary ul li img {
  margin-bottom:-0.5em;
  margin-top:0.5em;
  }
	  
  div.main {
  background-color: white;
  }
      
  div.module {
  padding-bottom:1.5em;
  padding-top:1.5em;
  }
	  
  div.footer {
  background-color: #EEE;
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 100%;
  font-weight: bold;
  width:100%;
  }


  a {
  color: #000080;
  }

  a:hover {
  color: #800000;
  }
      
  h2 {
  color: #800000;
  padding-bottom: 0;
  margin-bottom: 0;
  clear:left;
  }

  table { 
  margin-left: 3em;
  text-align: center;
  }
  
  th { 
  text-align: center;
  background-color: #000080;
  color: #FFF;
  padding: 0.4em;
  }      
  
  td { 
  font-family: monospace; 
  text-align: left;
  background-color: #EEE;
  color: #000;
  padding: 0.4em;
  }

  img {
  padding-top: 0;
  margin-top: 0;
  border-top: 0;
  }

  
  p {
  padding-top: 0;
  margin-top: 0;
  }
  
</style>

</head>
<body>
<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Sun 19 Jan 2014<br />
C2AFAACXX_Pc2_0d_13s004125-1-1_Hejnol_lane213s004125_sequence.txt.gz
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>C2AFAACXX_Pc2_0d_13s004125-1-1_Hejnol_lane213s004125_sequence.txt.gz</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>19240372</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>52</td>
</tr>
<tr>
<td>%GC</td>
<td>47</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/error.png" alt="[FAIL]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/error.png" alt="[FAIL]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GATCGGAAGAGCACACGTCTGAACTCCAGTCACGTGGCCTTATCTCGTATGC</td>
<td>389235</td>
<td>2.023011821185162</td>
<td>TruSeq Adapter, Index 1 (97% over 35bp)</td>
</tr>
<tr>
<td>AGATCGGAAGAGCACACGTCTGAACTCCAGTCACGTGGCCTTATCTCGTATG</td>
<td>194758</td>
<td>1.0122361459539349</td>
<td>TruSeq Adapter, Index 1 (97% over 35bp)</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/error.png" alt="[FAIL]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>CACGT</td>
<td>2386315</td>
<td>2.7980444</td>
<td>25.999386</td>
<td>14</td>
</tr>
<tr>
<td>CACAC</td>
<td>2399860</td>
<td>2.7543244</td>
<td>26.101974</td>
<td>12</td>
</tr>
<tr>
<td>GAAGA</td>
<td>2223065</td>
<td>2.3522415</td>
<td>23.433088</td>
<td>6</td>
</tr>
<tr>
<td>GCACA</td>
<td>1855095</td>
<td>2.1569476</td>
<td>26.337105</td>
<td>11</td>
</tr>
<tr>
<td>GTGGC</td>
<td>1565535</td>
<td>2.0973258</td>
<td>28.564373</td>
<td>34</td>
</tr>
<tr>
<td>ACACG</td>
<td>1783760</td>
<td>2.074005</td>
<td>25.918543</td>
<td>13</td>
</tr>
<tr>
<td>TTTTT</td>
<td>2362005</td>
<td>2.0493379</td>
<td>6.1021028</td>
<td>2</td>
</tr>
<tr>
<td>GTCAC</td>
<td>1729010</td>
<td>2.0273294</td>
<td>26.038033</td>
<td>29</td>
</tr>
<tr>
<td>GAGCA</td>
<td>1700730</td>
<td>2.0033324</td>
<td>27.229618</td>
<td>9</td>
</tr>
<tr>
<td>AAAAA</td>
<td>2348645</td>
<td>1.9538126</td>
<td>5.966785</td>
<td>2</td>
</tr>
<tr>
<td>AAAAT</td>
<td>2328810</td>
<td>1.9536781</td>
<td>5.9111524</td>
<td>3</td>
</tr>
<tr>
<td>GTCTG</td>
<td>1629440</td>
<td>1.9519238</td>
<td>26.501684</td>
<td>17</td>
</tr>
<tr>
<td>CTCGT</td>
<td>1649500</td>
<td>1.9504399</td>
<td>25.676628</td>
<td>44</td>
</tr>
<tr>
<td>GGAAG</td>
<td>1624850</td>
<td>1.938988</td>
<td>25.975252</td>
<td>5</td>
</tr>
<tr>
<td>ACGTG</td>
<td>1628800</td>
<td>1.9348123</td>
<td>25.507217</td>
<td>32</td>
</tr>
<tr>
<td>AGCAC</td>
<td>1609995</td>
<td>1.8719658</td>
<td>26.546238</td>
<td>10</td>
</tr>
<tr>
<td>CGTCT</td>
<td>1580745</td>
<td>1.8691411</td>
<td>25.997967</td>
<td>16</td>
</tr>
<tr>
<td>ACGTC</td>
<td>1586230</td>
<td>1.8599147</td>
<td>25.918064</td>
<td>15</td>
</tr>
<tr>
<td>TCACG</td>
<td>1578405</td>
<td>1.8507396</td>
<td>25.563856</td>
<td>30</td>
</tr>
<tr>
<td>TCCAG</td>
<td>1569845</td>
<td>1.8407027</td>
<td>26.15085</td>
<td>25</td>
</tr>
<tr>
<td>TCTCG</td>
<td>1545310</td>
<td>1.8272412</td>
<td>25.806856</td>
<td>43</td>
</tr>
<tr>
<td>TGAAC</td>
<td>1729630</td>
<td>1.8217643</td>
<td>23.238089</td>
<td>20</td>
</tr>
<tr>
<td>CGTGG</td>
<td>1356480</td>
<td>1.817258</td>
<td>28.273808</td>
<td>33</td>
</tr>
<tr>
<td>CTCCA</td>
<td>1566010</td>
<td>1.8124962</td>
<td>25.319893</td>
<td>24</td>
</tr>
<tr>
<td>AGAGC</td>
<td>1530050</td>
<td>1.8022841</td>
<td>26.548092</td>
<td>8</td>
</tr>
<tr>
<td>CTGAA</td>
<td>1687905</td>
<td>1.7778167</td>
<td>23.270403</td>
<td>19</td>
</tr>
<tr>
<td>CAGTC</td>
<td>1486315</td>
<td>1.7427605</td>
<td>25.493906</td>
<td>27</td>
</tr>
<tr>
<td>AAGAG</td>
<td>1642585</td>
<td>1.7380313</td>
<td>23.300661</td>
<td>7</td>
</tr>
<tr>
<td>CGGCG</td>
<td>1166865</td>
<td>1.725672</td>
<td>5.041301</td>
<td>1</td>
</tr>
<tr>
<td>CCAGT</td>
<td>1464250</td>
<td>1.7168882</td>
<td>25.435495</td>
<td>26</td>
</tr>
<tr>
<td>ATCTC</td>
<td>1624300</td>
<td>1.702999</td>
<td>22.964252</td>
<td>42</td>
</tr>
<tr>
<td>GAACT</td>
<td>1603825</td>
<td>1.6892579</td>
<td>23.133213</td>
<td>21</td>
</tr>
<tr>
<td>AACTC</td>
<td>1587480</td>
<td>1.6504524</td>
<td>22.806526</td>
<td>22</td>
</tr>
<tr>
<td>GCCTT</td>
<td>1389995</td>
<td>1.6435902</td>
<td>25.264017</td>
<td>37</td>
</tr>
<tr>
<td>GATCG</td>
<td>1337655</td>
<td>1.5889682</td>
<td>25.718643</td>
<td>1</td>
</tr>
<tr>
<td>TGGCC</td>
<td>1199665</td>
<td>1.5864227</td>
<td>27.633514</td>
<td>35</td>
</tr>
<tr>
<td>TCTGA</td>
<td>1491360</td>
<td>1.5840718</td>
<td>23.338234</td>
<td>18</td>
</tr>
<tr>
<td>AGTCA</td>
<td>1452720</td>
<td>1.5301038</td>
<td>22.853462</td>
<td>28</td>
</tr>
<tr>
<td>GTATG</td>
<td>1370755</td>
<td>1.4750152</td>
<td>23.100103</td>
<td>47</td>
</tr>
<tr>
<td>TCGTA</td>
<td>1357860</td>
<td>1.4422727</td>
<td>22.75621</td>
<td>45</td>
</tr>
<tr>
<td>ACTCC</td>
<td>1245105</td>
<td>1.4410816</td>
<td>25.097094</td>
<td>23</td>
</tr>
<tr>
<td>CGGAA</td>
<td>1211735</td>
<td>1.427333</td>
<td>25.388515</td>
<td>4</td>
</tr>
<tr>
<td>GGCCT</td>
<td>1070455</td>
<td>1.4155569</td>
<td>27.8902</td>
<td>36</td>
</tr>
<tr>
<td>TTATC</td>
<td>1490080</td>
<td>1.415218</td>
<td>20.4758</td>
<td>40</td>
</tr>
<tr>
<td>TCGGA</td>
<td>1183670</td>
<td>1.406053</td>
<td>25.64084</td>
<td>3</td>
</tr>
<tr>
<td>ATCGG</td>
<td>1170260</td>
<td>1.3901237</td>
<td>25.504768</td>
<td>2</td>
</tr>
<tr>
<td>TATCT</td>
<td>1417870</td>
<td>1.3466357</td>
<td>20.46613</td>
<td>41</td>
</tr>
<tr>
<td>GTTTT</td>
<td>1374805</td>
<td>1.3339896</td>
<td>5.243131</td>
<td>1</td>
</tr>
<tr>
<td>CTTAT</td>
<td>1351415</td>
<td>1.2835194</td>
<td>20.301802</td>
<td>39</td>
</tr>
<tr>
<td>CGTAT</td>
<td>1205375</td>
<td>1.2803082</td>
<td>22.68092</td>
<td>46</td>
</tr>
<tr>
<td>TATGC</td>
<td>1153935</td>
<td>1.2256706</td>
<td>22.812834</td>
<td>48</td>
</tr>
<tr>
<td>CCTTA</td>
<td>1142990</td>
<td>1.198369</td>
<td>22.213053</td>
<td>38</td>
</tr>
<tr>
<td>AGATC</td>
<td>1031785</td>
<td>1.0867463</td>
<td>11.385563</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
</body></html>